sdk/python/src/freewayai/prompt_formatted.py

```diff
@@ -1,9 +1,40 @@
+import copy, json
 from .prompt_messages_to_provider_format import convert_messages_to_format
+from .token_counting import count_messages_tokens
+from .utilities import _format_messages, _format_template_string
 
 class Prompt_Formatted:
-    def __init__(self, messages):
+    def __init__(self, template: dict, variables: dict):
+        self.template = template
+        self.variables = variables
+        self.messages = []
+
+
+        self.added_context = []
+        self._format_prompt()
+
+    def _format_prompt(self):
+
+        prompt = self.template
```

Comment:
I'm nitpicking here but why rename this to `prompt`? You probably don't need a local variable here at all given its only used a couple times below but at the very least I would think this would be `template = self.template`?